### DISCUSSION TITLE
Android SDK background api calls

### DISCUSSION THREAD

**Discussion by kazougagh** [GetStream/stream-chat-android#4593] [General]
Hi there,
I am seeing a lot of api call made by the android SDK. Can you explain why you need to make so many calls every 10 seconds ish ?
Thanks

`I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8729) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554583","cid":"*","created_at":"2022-12-12T15:47:13.651916685Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8727) {"type":"health.check","connection_id":"6315f3c9-0a05-2e4c-0000-000000554b2b","cid":"*","created_at":"2022-12-12T15:47:13.668997774Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8730) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554584","cid":"*","created_at":"2022-12-12T15:47:13.681337217Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8728) {"type":"health.check","connection_id":"6315f3c9-0a05-2e4c-0000-000000554b2c","cid":"*","created_at":"2022-12-12T15:47:13.69227201Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8731) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554585","cid":"*","created_at":"2022-12-12T15:47:13.698691275Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8729) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554583","cid":"*","created_at":"2022-12-12T15:47:24.700499805Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8727) {"type":"health.check","connection_id":"6315f3c9-0a05-2e4c-0000-000000554b2b","cid":"*","created_at":"2022-12-12T15:47:24.710798741Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8728) {"type":"health.check","connection_id":"6315f3c9-0a05-2e4c-0000-000000554b2c","cid":"*","created_at":"2022-12-12T15:47:24.732539063Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8730) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554584","cid":"*","created_at":"2022-12-12T15:47:24.715265973Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8731) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554585","cid":"*","created_at":"2022-12-12T15:47:24.73647957Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8729) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554583","cid":"*","created_at":"2022-12-12T15:47:35.773084545Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8727) {"type":"health.check","connection_id":"6315f3c9-0a05-2e4c-0000-000000554b2b","cid":"*","created_at":"2022-12-12T15:47:35.773485161Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8728) {"type":"health.check","connection_id":"6315f3c9-0a05-2e4c-0000-000000554b2c","cid":"*","created_at":"2022-12-12T15:47:35.77984425Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8730) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554584","cid":"*","created_at":"2022-12-12T15:47:35.780037982Z"}
I/Chat:Events: (OkHttp https://chat.stream-io-api.com/...:8731) {"type":"health.check","connection_id":"6315f4ad-0a05-4dc8-0000-000000554585","cid":"*","created_at":"2022-12-12T15:47:35.779980415Z"}`

**Comment by MarinTolic** [comment]
Hey kazougagh,

These are health check events which are used to confirm that the client connection is alive. They are a normal part of the SDK and working as intended.

**Comment by kazougagh** [comment]
Thanks @MarinTolic for the reply.
I understand the need but I think 5 health check calls every 10 sec (in my case) is a lot though.
Best.
